contrib/dimfeld/httptreemux.v5/httptreemux.go

```diff
@@ -115,17 +115,40 @@ func getRoute(router *httptreemux.TreeMux, w http.ResponseWriter, req *http.Requ
 
 	// Check for redirecting route due to trailing slash for parameters.
 	// The redirecting behaviour originates from httptreemux router.
-	if lr.StatusCode == http.StatusMovedPermanently && strings.HasSuffix(route, "/") {
-		rReq := req.Clone(req.Context())
-		rReq.RequestURI = strings.TrimSuffix(rReq.RequestURI, "/")
-		rReq.URL.Path = strings.TrimSuffix(rReq.URL.Path, "/")
-
-		lr, found = router.Lookup(w, rReq)
-		if !found {
-			return "", false
+	// TODO: what about the other two router.RedirectBehavior cases?
+	if lr.StatusCode == http.StatusMovedPermanently && router.RedirectBehavior == httptreemux.Redirect301 {
+		if strings.HasSuffix(route, "/") {
+			// if the route has a trailing slash, retry lookup without trailing slash
+			rReq := req.Clone(req.Context())
+			rReq.RequestURI = strings.TrimSuffix(rReq.RequestURI, "/")
+			rReq.URL.Path = strings.TrimSuffix(rReq.URL.Path, "/")
+
+			lr, found = router.Lookup(w, rReq)
+			if !found {
+				return "", false
+			}
+		} else {
+			// if the route does not have a trailing slash, retry lookup without trailing slash
+			rReq := req.Clone(req.Context())
+			rReq.RequestURI = rReq.RequestURI + "/"
+			rReq.URL.Path = rReq.URL.Path + "/"
+
+			lr, found = router.Lookup(w, rReq)
+			if !found {
+				return "", false
+			}
 		}
 	}
 
+	// remove trailing slash from route to standardize resource name
+	// the router does not allow you to register two matching routes with the only difference being a trailing slash
+	// this only affects the resource name and not the actual request URL set on tag http.url
+	routeLen := len(route)
+	trailingSlash := route[routeLen-1] == '/' && routeLen > 1
+	if trailingSlash {
+		route = strings.TrimSuffix(route, "/")
```

Comment:
This is to standardize the resource name.



For handler `GET /api/example` both of these cases will now have the same resource name:
- `GET /api/example` (without trailing slash)
- `GET /api/example/` (with trailing slash)

Without this change you can end up with a scenario where you have two different resources in Datadog:
1. `GET /api/example` (without trailing slash)
2. `GET /api/example/` (with trailing slash)

I believe this is a safe change to make because the router panics if you try and register two handlers with the same path with only the trailing slash being the difference, i.e.:
```golang
router := New()
router.GET("/api/example", handler200)
// this panics
router.GET("/api/example/", handler200)

func handler200(w http.ResponseWriter, _ *http.Request, _ map[string]string) {
	w.Write([]byte("OK\n"))
}
```